`flatten` does not work with `target_dim=1`

```
nested_ragged_tensors on  dev [$!] via  v3.11.9 via  NRT 
❯ pytest --doctest-modules -s --ignore=docs 
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.11.9, pytest-8.3.2, pluggy-1.5.0
rootdir: /home/mmd/Projects/nested_ragged_tensors
configfile: pyproject.toml
plugins: hydra-core-1.3.2, cov-5.0.0
collected 25 items                                                                                                                                                                           

sample_dataset_builder/src/sample_dataset_builder/dataset_generator.py .....
src/nested_ragged_tensors/ragged_numpy.py .........F..........

========================================================================================== FAILURES ==========================================================================================
______________________________________________________ [doctest] nested_ragged_tensors.ragged_numpy.JointNestedRaggedTensorDict.flatten ______________________________________________________
937                    [3, 3, 2, 2, 0, 0, 0]], dtype=uint8)
938             >>> dense_dict['val']
939             array([[1. , 0.2, 0. , 3.1, 0. , 1. , 2.2],
940                    [3. , 3.3, 2. , 0. , 0. , 0. , 0. ]], dtype=float32)
941             >>> J = JointNestedRaggedTensorDict({
942             ...     "ts": [1,         2,      3],
943             ...     "id": [[1, 2, 3], [3, 4], [1, 2]],
944             ...     "val": [[1.0, 0.2, 0.], [3.1, 0.], [1., 2.2]],
945             ... })
946             >>> dense_dict = J.flatten().to_dense()
UNEXPECTED EXCEPTION: KeyError('dim0/bounds')
Traceback (most recent call last):
  File "/home/mmd/mambaforge/envs/NRT/lib/python3.11/doctest.py", line 1355, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest nested_ragged_tensors.ragged_numpy.JointNestedRaggedTensorDict.flatten[12]>", line 1, in <module>
  File "/home/mmd/Projects/nested_ragged_tensors/src/nested_ragged_tensors/ragged_numpy.py", line 974, in flatten
    prev_bounds = self.tensors[f"dim{target_dim-1}/bounds"]
                  ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'dim0/bounds'
/home/mmd/Projects/nested_ragged_tensors/src/nested_ragged_tensors/ragged_numpy.py:946: UnexpectedException
================================================================================== short test summary info ===================================================================================
FAILED src/nested_ragged_tensors/ragged_numpy.py::nested_ragged_tensors.ragged_numpy.JointNestedRaggedTensorDict.flatten
================================================================================ 1 failed, 24 passed in 8.76s ================================================================================

```

--- a/src/nested_ragged_tensors/ragged_numpy.py
+++ b/src/nested_ragged_tensors/ragged_numpy.py
@@ -953,7 +953,7 @@ class JointNestedRaggedTensorDict:
             >>> J.flatten(dim=0)
             Traceback (most recent call last):
                 ...
-            ValueError: Only supports dim = -1 or 2 for now; got 0
+            ValueError: Only supports dim = -1 or 1 for now; got 0
         """
         if dim < 0:
             target_dim = self.max_n_dims + dim
@@ -971,7 +971,10 @@ class JointNestedRaggedTensorDict:
             for k in self.keys_at_dim(d):
                 out_tensors[f"dim{d}/{k}"] = self.tensors[f"dim{d}/{k}"]
 
-        prev_bounds = self.tensors[f"dim{target_dim-1}/bounds"]
+        if target_dim == 1:
+            prev_bounds = np.array([len(self)])
+        else:
+            prev_bounds = self.tensors[f"dim{target_dim-1}/bounds"]
         for d in range(target_dim, self.max_n_dims):
             curr_bounds = self.tensors[f"dim{d}/bounds"]
             st = np.concatenate([[0], curr_bounds[prev_bounds[:-1] - 1]])

--- a/src/nested_ragged_tensors/ragged_numpy.py
+++ b/src/nested_ragged_tensors/ragged_numpy.py
@@ -900,23 +900,6 @@ class JointNestedRaggedTensorDict:
         Examples:
             >>> J = JointNestedRaggedTensorDict({
             ...     "S": [1, 2],
-            ...     "id": [[[1, 2, 3], [3, 4], [1, 2]], [[3], [3, 2, 2]]],
-            ...     "val": [[[1.0, 0.2, 0.], [3.1, 0.], [1., 2.2]], [[3], [3.3, 2., 0]]],
-            ... })
-            >>> dense_dict = J.flatten().to_dense()
-            >>> dense_dict['S']
-            array([1, 2], dtype=uint8)
-            >>> dense_dict['id']
-            array([[1, 2, 3, 3, 4, 1, 2],
-                   [3, 3, 2, 2, 0, 0, 0]], dtype=uint8)
-            >>> dense_dict['val']
-            array([[1. , 0.2, 0. , 3.1, 0. , 1. , 2.2],
-                   [3. , 3.3, 2. , 0. , 0. , 0. , 0. ]], dtype=float32)
-            >>> dense_dict['dim1/mask']
-            array([[ True,  True,  True,  True,  True,  True,  True],
-                   [ True,  True,  True,  True, False, False, False]])
-            >>> J = JointNestedRaggedTensorDict({
-            ...     "S": [1, 2],
             ...     "ts": [[1,         2,      3],      [4,   5]],
             ...     "id": [[[1, 2, 3], [3, 4], [1, 2]], [[3], [3, 2, 2]]],
             ...     "val": [[[1.0, 0.2, 0.], [3.1, 0.], [1., 2.2]], [[3], [3.3, 2., 0]]],
@@ -955,6 +938,18 @@ class JointNestedRaggedTensorDict:
             >>> dense_dict['val']
             array([[1. , 0.2, 0. , 3.1, 0. , 1. , 2.2],
                    [3. , 3.3, 2. , 0. , 0. , 0. , 0. ]], dtype=float32)
+            >>> J = JointNestedRaggedTensorDict({
+            ...     "ts": [1,         2,      3],
+            ...     "id": [[1, 2, 3], [3, 4], [1, 2]],
+            ...     "val": [[1.0, 0.2, 0.], [3.1, 0.], [1., 2.2]],
+            ... })
+            >>> dense_dict = J.flatten().to_dense()
+            >>> dense_dict['ts']
+            array([1, 0, 0, 2, 0, 3, 0], dtype=uint8)
+            >>> dense_dict['id']
+            array([1, 2, 3, 3, 4, 1, 2], dtype=uint8)
+            >>> dense_dict['val']
+            array([1. , 0.2, 0. , 3.1, 0. , 1. , 2.2], dtype=float32)
             >>> J.flatten(dim=0)
             Traceback (most recent call last):
                 ...

